providers/build.info

```diff
@@ -66,6 +66,17 @@ SOURCE[$LIBFIPS]=$LIBCOMMON
 DEPEND[$LIBLEGACY]=$LIBCOMMON
 DEPEND[$LIBDEFAULT]=$LIBCOMMON
 
+#
+# load-balance provider stuff
+#
+# Currently, load-balance provider is built in, it means that libcrypto must
```

Comment:
Considering that you have libcrypto share the provctx structure contents with this load-balance provider, it *must* remain built-in.

The general rule is that providers *must not share* the definition of opaque structures.  However, built-in providers may "cheat" in this regard, because it's guaranteed that the structures are exactly the same on both sides of the libcrypto\<-\>provider edge.

Reply:
> The general rule is that providers _must not share_ the definition of opaque structures.

I understand this. The comments I put in build.info is not well thought-out. Sorry. I double checked the code of load-balance provider (lbprov.c). So far, it is safe to compile lbprov.c both as module and as static (built-in). I can provide this as a config option to the user in a later pull request, something similar to STATIC_LEGACY.